fuse_core/src/uuid.cpp

```diff
@@ -65,30 +59,47 @@ UUID generate()
   return uuid;
 }
 
-UUID generate(const std::string& namespace_string, const ros::Time& stamp)
+UUID generate(const std::string& namespace_string, const rclcpp::Time& stamp)
 {
-  constexpr size_t buffer_size = sizeof(stamp.sec) + sizeof(stamp.nsec);
+  // replicate ros1 behaviour for now
+  #warning "deprecated time API"
+  rcl_time_point_value_t stamp_val = stamp.nanoseconds();
+  int32_t stamp_s = stamp_val / (rcl_time_point_value_t)1e9;
+  int32_t stamp_ns = stamp_val % (rcl_time_point_value_t)1e9;
+
+  constexpr size_t buffer_size = sizeof(stamp_s) + sizeof(stamp_ns);
   std::array<unsigned char, buffer_size> buffer;
   auto iter = buffer.begin();
-  iter = std::copy(reinterpret_cast<const unsigned char*>(&stamp.sec),
-                   reinterpret_cast<const unsigned char*>(&stamp.sec) + sizeof(stamp.sec),
+
+  #warning "unsafe time packing"
+  iter = std::copy(reinterpret_cast<const unsigned char*>(&stamp_s),
+                   reinterpret_cast<const unsigned char*>(&stamp_s) + sizeof(stamp_s),
                    iter);
-  iter = std::copy(reinterpret_cast<const unsigned char*>(&stamp.nsec),
-                   reinterpret_cast<const unsigned char*>(&stamp.nsec) + sizeof(stamp.nsec),
+  iter = std::copy(reinterpret_cast<const unsigned char*>(&stamp_ns),
+                   reinterpret_cast<const unsigned char*>(&stamp_ns) + sizeof(stamp_ns),
                    iter);
   return generate(namespace_string, buffer.data(), buffer.size());
 }
 
-UUID generate(const std::string& namespace_string, const ros::Time& stamp, const UUID& id)
+UUID generate(const std::string& namespace_string, const rclcpp::Time& stamp, const UUID& id)
 {
-  constexpr size_t buffer_size = sizeof(stamp.sec) + sizeof(stamp.nsec) + UUID::static_size();
+  // replicate ros1 behaviour for now
+  #warning "deprecated time API"
+  rcl_time_point_value_t stamp_val = stamp.nanoseconds();
+  int32_t stamp_s = stamp_val / (rcl_time_point_value_t)1e9;
```

Comment:
Use [`RCUTILS_NS_TO_S`](https://github.com/ros2/rcutils/blob/c83c4c7b197e72527fc9e0b438f9f1e6225f70af/include/rcutils/time.h#L39). IIRC it avoids some weird behavior on some compilers.

Reply:
There's no RCUTILS macro for getting the nanoseconds section, but I tried to use similar logic.

https://github.com/locusrobotics/fuse/pull/281/commits/4473a1d7231941e2b5081041e229405b0f398552

Reply:
https://github.com/locusrobotics/fuse/pull/283/files#diff-4d7269f5ad306527f06d77e64f5230b27087a3be72516585a9e0dc9286db6116

Obsoleted!